src/main.cpp

```diff
@@ -60,34 +60,22 @@ std::optional<config_t> build_config(const std::string& file)
 		}
 	}
 
-	// free, fixed
+	// initial values
 	{
-		std::map<std::string, bool> initial_states;
+		std::map<std::string, float> initial_states;
 		data["initial_states"].get_to(initial_states);
 
-		for (const auto& initial_state : initial_states)
-		{
-			auto it = std::find(node_names.begin(), node_names.end(), initial_state.first);
-
-			if (it == node_names.end())
-			{
-				std::cout << "Nonexisting node in initial_states part of config file" << std::endl;
-				return std::nullopt;
-			}
-
-			if (initial_state.second)
-			{
-				int bit = (int)std::distance(node_names.begin(), it);
-				config.fixed_part.set(bit);
-			}
-		}
-
+		config.initial_values.resize(node_names.size());
 		for (int i = 0; i < node_names.size(); i++)
 		{
 			auto it = initial_states.find(node_names[i]);
-
-			if (it == initial_states.end())
-				config.free_mask |= state_t(i);
+			
+			if (it == initial_states.end()) {
+				config.initial_values[i] = 0.5f;
+			} else {
+				config.initial_values[i] = it->second;
+			}
+		
 		}
```

Comment:
if you changed this `for` to iterate over initial_values instead of nodes, you could also check if user specified nonexisting node name.

start by `config.initial_values.resize(node_names.size(), 0.5f);` which will set defaults to all and then, in the for loop over initial_states, update only those nodes's values that are in initial_states (similarly as is done for internals)

Reply:
also please use clang-format (vscode: F1 -> Format Document) to have unified braces formatting (other files as well)